samples/rust/rust_alloc_tests.rs

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-2.0
+
+//! Rust alloc test.
+
+use alloc::alloc::{alloc, dealloc};
+use core::alloc::Layout;
+use core::ptr;
+use kernel::prelude::*;
+
+module! {
+    type: RustAllocTests,
+    name: b"rust_alloc_tests",
+    author: b"Rust for Linux Contributors",
+    description: b"Rust alloc tests",
+    license: b"GPL",
+}
+
+struct RustAllocTests;
+
+impl kernel::Module for RustAllocTests {
+    fn init(_name: &'static CStr, _module: &'static ThisModule) -> Result<Self> {
+        pr_info!("Rust alloc tests (init)\n");
+
+        {
+            let mut unaligned = false;
+            let mut min_unaligned = [(10000, ptr::null_mut()); 13];
+            let mut max_unaligned = [(0, ptr::null_mut()); 13];
+
+            // Generate size [1, 10000) and alignment [1,2,..,4096]
+            for size in 1..10000 {
+                for align in 0..13 {
+                    let layout = Layout::from_size_align(size, 1 << align)?;
+
+                    // SAFETY: layout.size() is not zero.
+                    let ptr = unsafe { alloc(layout) };
+
+                    if !ptr.is_null() {
```

Comment:
An allocator that always fails to allocate would pass the test. Perhaps you want to note it here?

Reply:
Yes, I probably should mention that this test is only for alignment requirement (i.e. add a line of comment before the test block).